phoenix-hive/src/main/java/org/apache/phoenix/hive/PhoenixSerde.java

```diff
@@ -0,0 +1,169 @@
+                    this.fieldOIs);
+        this.row = new ArrayList(this.columnNames.size());
+    }
+    
+    
+    /**
+     * This Deserializes a result from Phoenix to a Hive result
+     * @param wr the phoenix writable Object here PhoenixHiveDBWritable
+     * @return  Object for Hive
+     */
+
+    public Object deserialize(Writable wr) throws SerDeException {
+        if (!(wr instanceof PhoenixHiveDBWritable)) throw new SerDeException(
+                "Serialized Object is not of type PhoenixHiveDBWritable");
+        try {
+            this.row.clear();
+            PhoenixHiveDBWritable phdbw = (PhoenixHiveDBWritable) wr;
+            for (int i = 0; i < this.fieldCount; i++) {
+                Object value = phdbw.get((String) this.columnNames.get(i));
+                if (value != null) this.row.add(HiveTypeUtil.SQLType2Writable(
+                    ((TypeInfo) this.columnTypes.get(i)).getTypeName(), value));
+                else {
+                    this.row.add(null);
+                }
+            }
+            return this.row;
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new SerDeException(e.getCause());
+        }
+    }
+
+    public ObjectInspector getObjectInspector() throws SerDeException {
+        return this.ObjectInspector;
+    }
+
+    public SerDeStats getSerDeStats() {
+        return null;
+    }
+
+    /**
+     * This is a getter for the  serialized class to use with this SerDE
+     * @return  The class PhoenixHiveDBWritable
+     */
+    
+    public Class<? extends Writable> getSerializedClass() {
+        return PhoenixHiveDBWritable.class;
+    }
+
+    
+    /**
+     * This serializes a Hive row to a Phoenix entry
+     * incoming hive types.
+     * @param row Hive row
+     * @param inspector inspector for the Hive row
+     */
+    
+    public Writable serialize(Object row, ObjectInspector inspector) throws SerDeException {
+        final StructObjectInspector structInspector = (StructObjectInspector) inspector;
+        final List<? extends StructField> fields = structInspector.getAllStructFieldRefs();
+
+        if (fields.size() != fieldCount) {
+            throw new SerDeException(String.format("Required %d columns, received %d.", fieldCount,
+                fields.size()));
+        }
+        phrecord.clear();
+        for (int i = 0; i < fieldCount; i++) {
+            StructField structField = fields.get(i);
+            if (structField != null) {
+                Object field = structInspector.getStructFieldData(row, structField);
+                ObjectInspector fieldOI = structField.getFieldObjectInspector();
+                switch (fieldOI.getCategory()) {
+                case PRIMITIVE:
+                    Writable value =
+                            (Writable) ((PrimitiveObjectInspector) fieldOI)
+                                    .getPrimitiveWritableObject(field);
+                    phrecord.add(value);
```

Comment:
Will this write the data in a Phoenix serialized format? This is for data coming in from Hive?